user-impl/src/main/java/com/example/auction/user/impl/UserServiceImpl.java

```diff
@@ -61,11 +62,34 @@ public UserServiceImpl(PersistentEntityRegistry registry, UserRepository userRep
         return req -> userRepository.getUsers(pageNo.orElse(0), pageSize.orElse(DEFAULT_PAGE_SIZE));
     }
 
+    @Override
+    public ServiceCall<UserRegistration, String> login() {
+        return (UserRegistration req) -> {
+            return credRef(req.getEmail()).ask(PUserCommand.GetPUser.INSTANCE)
+                    .thenApply(maybeUser-> {
+                                if(maybeUser.isPresent()) {
+                                    if (PUserCommand.checkPassword(req.getPassword(),maybeUser.get().getPasswordHash())){
+                                        return maybeUser.get().getId().toString();
+                                    } else {
+                                        throw new NotFound("Email or password does not match ");
+                                    }
+                                } else {
+                                    throw new NotFound("User not found");
+                                }
+
+                            }
+                    );
+        };
+    }
     private PersistentEntityRef<PUserCommand> entityRef(UUID id) {
         return entityRef(id.toString());
     }
 
     private PersistentEntityRef<PUserCommand> entityRef(String id) {
         return registry.refFor(PUserEntity.class, id);
     }
+
+    private PersistentEntityRef<PUserCommand> credRef(String email) {
+        return registry.refFor(PUserEntity.class, email);
+    }
```

Comment:
I'm not sure what is the intent of this but you can't have two instances for the same User (a first instance referenced by the `userId` and a second instance referenced by the `email`).

```
     private PersistentEntityRef<PUserCommand> entityRef(String id) {
         return registry.refFor(PUserEntity.class, id);
     }
    private PersistentEntityRef<PUserCommand> credRef(String email) {
        return registry.refFor(PUserEntity.class, email);
    }
```